project/src/main/java/com/google/autograder/servlets/GetAssignmentDetailsServlet.java

```diff
@@ -0,0 +1,41 @@
+package com.google.autograder.servlets;
+
+import java.net.URL;
+import java.io.IOException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Entity;
+import com.google.autograder.servlets.helpers.Services;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+
+@WebServlet("/getAssignmentDetails")
+public final class GetAssignmentDetailsServlet extends HttpServlet {
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String requestURL = (String) request.getHeader("Referer");
+        String courseID = requestURL.substring(requestURL.indexOf("?courseID=") + 10);
+        String assignmentID = requestURL.substring(requestURL.indexOf("?assignmentID=") + 14, requestURL.indexOf("&"));
+
+        Filter assignmentIDFilter = new FilterPredicate("id", FilterOperator.EQUAL, assignmentID);
+        Filter courseIDFilter = new FilterPredicate("courseId", FilterOperator.EQUAL, courseID);
+
+        Query query = new Query("Assignment").setFilter(courseIDFilter).setFilter(assignmentIDFilter);
+        PreparedQuery results = Services.DATA_STORE.prepare(query);
+        Entity assignment = results.asIterable().iterator().next();
```

Comment:
What if there's no results? (and similar for the other servlets querying from datastore)

Reply:
This page can only be accessed if the assignment is already in the Datastore. If not then the link to this page is never displayed.

Reply:
The page can be accessed if someone types in the URL, in that case this should also return an error to the client. Similar to above, it's fine to leave for now with a TODO to handle this case

Reply:
Alright, I have the TODO there for now. I think that some of the changes I made with the Database in later PRs will make this work easier.